tfhe/src/core_crypto/commons/generators/encryption.rs

```diff
@@ -45,6 +45,15 @@ impl<G: ByteRandomGenerator> EncryptionRandomGenerator<G> {
         self.mask.remaining_bytes()
     }
 
+    pub(crate) fn fork_n(
+        &mut self,
+        n: usize,
+    ) -> Result<impl Iterator<Item = EncryptionRandomGenerator<G>>, ForkError> {
+        let mask_bytes = self.mask.remaining_bytes().unwrap() / n;
+        let noise_bytes = self.noise.remaining_bytes().unwrap() / n;
```

Comment:
as Thomas indicated, translate a remaining_bytes error to a ForkError

Reply:
So the enum comes from csprng and we have three variants

```rust
pub enum ForkError {
    ForkTooLarge,
    ZeroChildrenCount,
    ZeroBytesPerChild,
}
```

I used ForkTooLarge but not sure it's the best thing as what can fail on our end is a conversion from a u128 that is too big into a usize

Reply:
@tmontaigu for the forkerror we discussed at some point does the reasoning above sound ok ?

Reply:
Yes it seems reasonable